### DISCUSSION TITLE
Metamask connect issue

### DISCUSSION THREAD

**Discussion by Javier-Szyfer** [wevm/wagmi#206] [Q&A]
Hey guys, everytime i attempt to connect via metamask the login prompt doesn't show and instead a notification on the MM icon appears. On the console the following error is thrown : "Already processing eth_requestAccounts. Please wait."

Is there a way to always make the MM login prompt to reveal when someone press connect? i assume this error is because is there is an  unresolved promise. But if there  is any workarround this issue i would appreciate it.

Thanks

**Comment by tmm** [comment]
There's not much you can do here. If someone already has the MM prompt open, you can't trigger another request to open it.

You could disable the "Connect Wallet" button if `useConnect`'s `loading` property is `true` or change the text on the button to "Check Wallet".
